superset/views/core.py

```diff
@@ -1229,7 +1229,10 @@ def datasource_info(datasource_id, datasource_type, form_data):
         datasource = form_data.get('datasource', '')
         if '__' in datasource:
             datasource_id, datasource_type = datasource.split('__')
-        datasource_id = int(datasource_id)
+            # The case where the datasource has been deleted
+            datasource_id = None if datasource_id == 'None' else datasource_id
```

Comment:
I don't understand how `datasource_id == 'None'` comes into being in the first place. Can we catch the issue upstream?

Reply:
True. I was also confused at the beginning. `datasource` evaluates to 'None__Table', so splitting that gets `'None'`. I'll investigate upstream.